### PR TITLE
Fixed the shuttle navigation computer allowing you to move docking ports

### LINKED ISSUE #31486: The white ship on meta is very bugged.
Issue reported from Round ID: 77188 (/tg/Station Bagil [ENGLISH] [100% FREE LAG])

[Directions]: # (Include the Round ID from the Status panel or retrieve it from https://atlantaned.space/newSS13tools/round.php ! If you believe the issue to be caused by a testmerge [OOC tab -> Show Server Revision], report it in the pull request's comment section instead. Explain your issue in detail, including the steps to reproduce it.)

[For Admins]: # (Oddities induced by var-edits and other admin tools are not necessarily bugs. Verify that your issues occur under regular circumstances before reporting them.)

The white ship on meta is bugging out, strips of the sides just get left behind at times, the shuttle windows and doors might just disappear, Or the whole ship will bug out and gib everyone inside it.

### PR DESCRIPTION
**Pull request by Cruix**
Moving any shuttle to a custom location that included a docking port would cause that docking port to be moved with the shuttle next time it left. With some finagling you could even move the custom docking port to transit space.
Fixes #31486

### CODE REVIEW

**Comment by ninjanomnom** [comment]
This is the wrong way to fix it, instead make it so custom dockers can't move their shuttle into the region of an existing stationary dock.

**Comment by Cruix** [comment]
That is the case right now, with the exception of their own dock and the ones they can jump to. Removing that would be a huge decline in quality of life.

File: code/modules/shuttle/on_move.dm [hunk 1]
```diff
@@ -324,21 +324,22 @@ All ShuttleMove procs go here
...
+/atom/movable/light/onShuttleMove(turf/newT, turf/oldT, rotation, list/movement_force, move_dir, obj/docking_port/stationary/old_dock, obj/docking_port/mobile/moving_dock)
 	return FALSE
 
-/obj/docking_port/stationary/onShuttleMove(turf/newT, turf/oldT, rotation, list/movement_force, move_dir, old_dock)
+/obj/docking_port/stationary/onShuttleMove(turf/newT, turf/oldT, rotation, list/movement_force, move_dir, obj/docking_port/stationary/old_dock, obj/docking_port/mobile/moving_dock)
 	if(old_dock == src) //Don't move the dock we're leaving
 		return FALSE
-
+	if(moving_dock && !moving_dock.can_move_docking_ports)
+		return FALSE
```
Comment (ninjanomnom) [inline code/modules/shuttle/on_move.dm]: No need to check for moving_dock existing, if it doesn't exist I want that error to occur.

`if(!moving_dock.can_move_docking_ports || old_dock == src)`

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by duncathan — CHANGES_REQUESTED** [review]
per ninja's request

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**